CLKernel.cpp

```diff
@@ -377,6 +377,7 @@ void CLKernel::run(cl_command_queue *queue, int ND, const size_t *global_ws, con
     for(int i = 0; i < (int)wrappersToDirty.size(); i++) {
         wrappersToDirty[i]->markDeviceDirty();
     }
+    cl->finish();
```

Comment:
It's a good idea to request all the buffers in one go, and then call cl->finish().  On the whole, I'd prefer not to have additional sync points, when they're not needed. can we make this:
- either only sync if there are at least one outputbuffer? or...
- have an option, so by default we conserve the current behavior, but by flipping some switch somewhere, it will always sync?